RoaringBitmap/src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java

```diff
@@ -27,6 +27,28 @@
 // @Beta: this class is still in early stage. Its API may change and has not proofed itself as
 // bug-proof
 public class Roaring64NavigableMap implements Externalizable, LongBitmapDataProvider {
+  // The initial format of Roaring64NavigableMap
+  // 1: a boolean indicating if we process signed or unsigned longs
+  // 2: an int indicating the umber of highToLow buckets
+  // 3: loop over buckets: an int representating the high part,
+  // then the lowParts as standard RoaringBitmaps format
+  public static final int SERIALIZATION_MODE_LEGACY = 0;
+
+  // The format of Roaring64NavigableMap as defined in
+  // https://github.com/RoaringBitmap/RoaringFormatSpec#extention-for-64-bit-implementations
+  // It is compatible with CRoaring and GoRoaring
+  // https://github.com/RoaringBitmap/CRoaring/blob/master/cpp/roaring64map.hh#L959
+  // https://github.com/RoaringBitmap/CRoaring/blob/master/cpp/roaring64map.hh#L991
+  // https://github.com/RoaringBitmap/CRoaring/commit/efcb83dcdf332f02cde058a48574f5b7b14f73fb
+  // 1: a boolean indicating if we process signed or unsigned longs
+  // 2: an int indicating the umber of highToLow buckets
+  // 3: loop over buckets: an int representating the high part,
+  // then the lowParts as standard RoaringBitmaps format
+  public static final int SERIALIZATION_MODE_PORTABLE = 1;
+
+  // As of RoaringBitmap 0.X, we stick to the legacy format for retrocompatibility
+  // RoaringBitmap 1.X shall switch to the portable format by default
```

Comment:
I cannot see how to do it, but in an ideal world, we would warn at compile-time the programmers that something is up with `SERIALIZATION_MODE`.

Reply:
I do not get your point. In your dreams, what would like like to see/report/behave?

Reply:
In my dream, people building the code would see 

> I see that you are using the class `Roaring64NavigableMap` with `SERIALIZATION_MODE = SERIALIZATION_MODE_LEGACY`, and you read and write bitmaps. Please be advised that the serialization format might change in the future.

I don't think that's actually possible in Java (or in a language that I know?).

Reply:
I raised the concern because even if I cannot see a solution, some people might have an idea.

Reply:
First, by marking SERIALIZATION_MODE_LEGACY as deprecated, it may lead to a compile-time WARN. We may need to push this constant out of Roaring64NavigableMap, and use it through it through a static initializerblock to force the compile warning about this deprecated usage. (I'm no expert in this mechanism, but it exists).

As in:

```
/Users/...RoaringBitmap/RoaringBitmap/src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java:8: warning: [deprecation] MutableRoaringBitmapPrivate in org.roaringbitmap.buffer has been deprecated
import org.roaringbitmap.buffer.MutableRoaringBitmapPrivate;
```

Else, a fancy solution may be done à la ErrorProne, which reports code-issues at compilation-time. (I'm not suggest doing that, but it does exists). (By the way, I may hook ErrorProne in the project, to make sure it does not detect anything.)